### PR TITLE
Added TLS backend selection features & `Default`, removed `get_`/`Get` prefixes

### PR DESCRIPTION
**Pull request by virtualritz**
The main reason is that `request`'s default TLS dependency pulls in `openssl` which wraps the OpenSSL C-lib via `openssl-sys`.
This not only adds a noticable hiccup to build times, it also requires a resp. C compiler toolchain which can be a PITA when cross-compilation for other platform targets is needed (e.g. Android or iOS).

`reqwest` offers [a bunch of alternative TLS backends](https://docs.rs/reqwest/latest/reqwest/#optional-features), exposed by feature flags.

With this PR they're all also available in the `infobip_sdk` crate.

This PR also puts some code behind `#[cfg(feature=...)]` gates to get rid of some warnings and further decrease build time.

### CODE REVIEW

**Comment by virtualritz** [comment]
That spellcheck will never succeed btw. for obvious reasons.

**Comment by virtualritz** [comment]
This now also fixes #13 and #16. There are some minor changes for developer UX too. I.e. releasing this requires at least a bump to `v0.6`.

File: Cargo.toml [hunk 1]
```diff
@@ -7,29 +7,48 @@ homepage = "https://www.infobip.com/"
...
+whatsapp = []
+# See https://docs.rs/reqwest/latest/reqwest/#optional-features
+## Enables TLS functionality provided by `native-tls`.
+native-tls = ["reqwest/native-tls"]
+## Enables the `vendored` feature of `native-tls`.
+native-tls-vendored = ["reqwest/native-tls-vendored"]
+# Enables the `alpn` feature of `native-tls`.
+#native-tls-alpn = [ "reqwest/native-tls-alpn" ]
+## Enables TLS functionality provided by `rustls`. Equivalent to `rustls-tls-webpki-roots`.
+rustls-tls = ["reqwest/rustls-tls"]
```
Comment (ukmadlz) [inline Cargo.toml:29]: Could you add `rustls` to the `./.cspell.json` file for the spell check test?

Comment (virtualritz) [reply Cargo.toml:29]: Done.

**Review by ukmadlz — APPROVED** [review]

**Pull request merged**